console/console-init/ui/src/modules/address-space/dialogs/EditAddressSpaceContainer/EditAddressSpaceContainer.tsx

```diff
@@ -55,13 +55,12 @@ export const EditAddressSpaceContainer: React.FunctionComponent<{}> = () => {
     if (addressSpace.type) {
       planOptions =
         addressSpacePlans
+          .filter(plan => plan.spec.addressSpaceType === addressSpace.type)
           .map(plan => {
-            if (plan.spec.addressSpaceType === addressSpace.type) {
-              return {
-                value: plan.metadata.name,
-                label: plan.metadata.name
-              };
-            }
+            return {
+              value: plan.metadata.name,
+              label: plan.metadata.name
+            };
           })
           .filter(plan => plan !== undefined) || [];
```

Comment:
can you try to move this filter check in above filter.
addressSpacePlans.filter(plan => plan.spec.addressSpaceType === addressSpace.type)==>
addressSpacePlans.filter(plan =>plan !== undefined &&  plan.spec.addressSpaceType === addressSpace.type)

Reply:
Done